### DISCUSSION TITLE
How Long Before I Can Reuse a Repository Name

### DISCUSSION THREAD

**Discussion by dknapp67** [community/community#131258] [Repositories] [answered]
### Select Topic Area

Question

### Body

I've deleted a repository. I am trying to create a repository from my Debian box, and keep receiving the error "remote origin already exists." I currently have no repositories on my github profile. How long before I can use that name again?

**Comment by davevad93** [comment]
Hi @dknapp67 ,

I think about an hour, but I'm not totally sure, I say that because that's the waiting time needed to restore a deleted repo as stated in the [docs](https://docs.github.com/en/repositories/creating-and-managing-repositories/restoring-a-deleted-repository):

- ***It can take up to an hour after a repository is deleted before that repository is available for restoration.***

So I guess that's how long it takes for GitHub to list a repo as deleted.

**Reply by dknapp67** [reply]
Well, 24 hours and I am still receiving the error.

I am using the following command:
`git remote add origin https://github.com/dknapp67/zero2prod.git`

**Reply by davevad93** [reply]
Hey @dknapp67 that's odd, first of all make sure the remote and its references have been properly deleted by running `git remote -v`, if the repo origin is still listed, remove it by running `git remote rm origin`, run `git remote -v` again to make sure it's gone, you should now be able to run `git remote add origin` without any issues.

For more info, here's the official documentation about [Managing remote repositories](https://docs.github.com/en/get-started/getting-started-with-git/managing-remote-repositories#removing-a-remote-repository).

**Comment by princeajuzie7** [comment, marked as answer]
Here's how you can address this issue you are facing:

1. **Check the current remote repository:**
   You can check which remote repositories are currently configured in your local Git repository by running:
   ```sh
   git remote -v
   ```

2. **Remove the existing remote origin:**
   If you see an existing remote origin that you don't need, you can remove it using:
   ```sh
   git remote remove origin
   ```

3. **Add the new remote origin:**
   After removing the old remote origin, you can add a new one using:
   ```sh
   git remote add origin https://github.com/username/repository-name.git
   ```

Regarding the repository name availability on GitHub:

- Once you delete a repository from GitHub, the name becomes immediately available for reuse. You can create a new repository with the same name as soon as the deletion process is complete.

Here is a step-by-step guide to create a new repository and link it to your local directory:

1. **Create a new repository on GitHub:**
   Go to your GitHub profile, click on "New" to create a new repository, and give it the desired name.

2. **Initialize your local directory as a Git repository (if it's not already):**
   ```sh
   git init
   ```

3. **Add your new remote repository:**
   ```sh
   git remote add origin https://github.com/username/repository-name.git
   ```

4. **Add and commit your files:**
   ```sh
   git add .
   git commit -m "Initial commit"
   ```

5. **Push your changes to GitHub:**
   ```sh
   git push -u origin master
   ```

By following these steps, you should be able to resolve the "remote origin already exists" error and successfully link your local repository to the new GitHub repository.

hope it was helpful

**Reply by dknapp67** [reply]
@princeajuzie7 This did it for me. Thank you!

**Reply by rudimusmaximus** [reply]
this does not seem to be the case when using the cli, i get "GraphQL: Name already exists on this account (createRepository)". So, i created a local repo and after a few commits mistakenly setup the new repo in my personal account, so i deleted it and removed the remote. Now when i attempt, "gh repo create intendedorg/reusedlocalname --private --source=. --remote=origin".

**Reply by rudimusmaximus** [reply]
> this does not seem to be the case when using the cli, i get "GraphQL: Name already exists on this account (createRepository)". So, i created a local repo and after a few commits mistakenly setup the new repo in my personal account, so i deleted it and removed the remote. Now when i attempt, "gh repo create intendedorg/reusedlocalname --private --source=. --remote=origin".

anyone, else experience this with the gh cli? I'll try again in a bit.

**Comment by princeajuzie7** [comment]
hey  @dknapp67 am happy  it helped.
